src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultComplexObjectValidationStrategy.cs

```diff
@@ -84,36 +84,20 @@ public bool MoveNext()
                 var propertyName = property.BinderModelName ?? property.PropertyName;
                 var key = ModelNames.CreatePropertyModelName(_key, propertyName);
 
-                object model;
-
-                // Our property accessors don't work on Mono 4.0.4 - see https://github.com/aspnet/External/issues/44
-                // This is a workaround for what the PropertyGetter does in the background.
-                if (IsMono)
+                if (_model == null)
+                {
+                    // Performance: Never create a delegate when container is null.
+                    _entry = new ValidationEntry(property, key, model: null);
+                }
+                else if (IsMono)
                 {
-                    if (_model == null)
-                    {
-                        model = null;
-                    }
-                    else
-                    {
-                        var propertyInfo = _model.GetType().GetRuntimeProperty(property.PropertyName);
-                        try
-                        {
-                            model = propertyInfo.GetValue(_model);
-                        }
-                        catch (TargetInvocationException ex)
-                        {
-                            throw ex.InnerException;
-                        }
-                    }
+                    _entry = new ValidationEntry(property, key, () => GetModelOnMono(_model, property.PropertyName));
```

Comment:
The `Model` accesses fail everywhere.

But, I now suspect you were asking specifically about the Mono workaround. We have repeatedly chosen to leave all similar workarounds alone. I wasn't going to spend the time checking whether the problem has been fixed.